2nd-gen/auth-blocking-functions/functions/index.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const {beforeUserCreated, beforeUserSignedIn, HttpsError} = require("firebase-functions/v2/identity");
+const {admin} = require("firebase-admin");
+
+admin.initializeApp();
+const store = admin.firestore();
+
+// [START v2BlockCreate]
+// [START v2beforeCreateFunctionTrigger]
+// Block account creation with any non-gmail email address.
+exports.blockcreate = beforeUserCreated((event) => {
+  // [END v2beforeCreateFunctionTrigger]
+  // [START v2readEmailData]
+  // Email passed from the CloudEvent.
+  const email = event.data.email || "";
+  // [END v2readEmailData]
+
+  // [START v2domainHttpsError]
+  // Checking that the email is a 'gmail' domain.
+  if (!email?.includes("gmail.com")) {
```

Comment:
```suggestion
  // Only users of a specific domain can sign up.
  if (!email?.includes("@acme.com")) {
```

update to match the docs [here](https://firebase.google.com/docs/functions/beta/auth-blocking-events#blocking_registration_or_sign-in)